engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -320,14 +319,14 @@ class EngineJobExecutionActor(replyTo: ActorRef,
     ()
   }
 
-  def makeFetchCachedResultsActor(metaInfoId: MetaInfoId, taskOutputs: Seq[TaskOutput]): Unit = {
-    context.actorOf(FetchCachedResultsActor.props(metaInfoId, self, new CallCache(SingletonServicesStore.databaseInterface)))
+  def makeFetchCachedResultsActor(callCachingEntryId: CallCachingEntryId, taskOutputs: Seq[TaskOutput]): Unit = {
+    context.actorOf(FetchCachedResultsActor.props(callCachingEntryId, self, new CallCache(SingletonServicesStore.databaseInterface)))
     ()
   }
 
-  private def fetchCachedResults(taskOutputs: Seq[TaskOutput], metaInfoId: MetaInfoId) = {
-    makeFetchCachedResultsActor(metaInfoId, taskOutputs)
-    goto(FetchingCachedOutputsFromDatabase)
+  private def fetchCachedResults(taskOutputs: Seq[TaskOutput], callCachingEntryId: CallCachingEntryId, data: ResponsePendingData) = {
+    makeFetchCachedResultsActor(callCachingEntryId, taskOutputs)
+    goto(FetchingCachedOutputsFromDatabase) using data
```

Comment:
Yes so the idea is that if data.cacheHit is `Some(NonEmptyList)`, then the head is the current cacheHit being processed. We need this because when we get the response from the backend copying cache outputs, it doesn't send back the cache hit ID, so we need to remember it in case it failed and we want to invalidate it.
At first I thought of the data.cacheHit value to mean `if the current cacheHit fails, these are the potential next ones to try`, but by doing this we would lose the information about what the current one actually is, hence the above change.